server/src/session.ts

```diff
@@ -49,6 +49,12 @@ export class Session {
   private readonly logToConsole: boolean;
   private diagnosticsTimeout: NodeJS.Timeout|null = null;
   private isProjectLoading = false;
+  /**
+   * By default, the rename feature is enabled. However, if we detect the compiler options
+   * diagnostic that suggests enabling strict mode, we want to disable renaming because we know that
+   * there are many cases where it will not work correctly.
+   */
+  private disableRenaming = false;
```

Comment:
This should be a per-project setting, so we probably want a `WeakMap<ts.server.Project, boolean>` here.

Reply:
Good catch - I made this a `Set<ts.server.Project>` to just track which projects have renaming disabled.